File: infrastructures/ddd/typescript/src/api/dto/v1/requests/UpdateExampleRequest.ts
import { BaseRequestDto } from '../infra/BaseRequestDto';

/**
 * Request DTO for updating an existing example.
 * Request DTOs:
 * - Live in /api/dto/v1/requests/
 * - Suffix with "Request"
 * - Used for HTTP request payloads
 * - Transport only - NO business logic
 * - Validate at edge (API layer)
 */
export class UpdateExampleRequest extends BaseRequestDto {
  /**
   * The updated name of the example
   * Required, max 100 characters
   */
  public Name: string;

  /**
   * The updated description of the example
   * Optional
   */
  public Description: string;

  constructor() {
    super();
    this.Name = '';
    this.Description = '';
  }
}

File: infrastructures/ddd/csharp/src/Api/Dto/V1/Requests/UpdateExampleRequest.cs
using System.ComponentModel.DataAnnotations;
using OddlyDdd.Api.Dto.V1.Infra;

namespace OddlyDdd.Api.Dto.V1.Requests
{
    /// <summary>
    /// Example request DTO for updating an example.
    /// </summary>
    public class UpdateExampleRequest : BaseRequestDto
    {
        [Required(ErrorMessage = "Name is required")]
        [StringLength(100, ErrorMessage = "Name cannot exceed 100 characters")]
        public string Name { get; set; } = string.Empty;

        [StringLength(500, ErrorMessage = "Description cannot exceed 500 characters")]
        public string Description { get; set; } = string.Empty;
    }
}
